libqpdf/QPDF.cc

```diff
@@ -2034,13 +2034,13 @@ QPDF::copyForeignObject(QPDFObjectHandle foreign)
     }
     obj_copier.to_copy.clear();
 
-    auto& result = obj_copier.object_map[foreign.getObjGen()];
-    if (!result.isInitialized()) {
-        result = QPDFObjectHandle::newNull();
+    auto og = foreign.getObjGen();
+    if (!obj_copier.object_map.count(og)) {
         warn(damagedPDF("Unexpected reference to /Pages object while copying foreign object. "
```

Comment:
I missed this before, but to be consistent with the style of most warnings, you could you change this to lower-case and semicolon-separated, as in `unexpected reference to /Pages object while copying foreign object;  replacing with null`